### PR TITLE
remove rust-cypto dependency, fix issuse 2404 3254

### PR DESCRIPTION
**Pull request by nkysg**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe):

## What is the current behavior?
fix issue
https://github.com/starcoinorg/starcoin/issues/2404
https://github.com/starcoinorg/starcoin/issues/3254 

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Review by JerryKwan — COMMENTED** [review]
@nkysg 
Seems like you are using tiny-keccak to replace rust-crypto, but the latest commit to tini-keccak is 2 years ago. Not sure if it is maintained.
When reading the codes in Move language, we noticed that they are using crate sha2 to do some hash algorithms.
image

https://docs.rs/sha2/latest/sha2/
https://github.com/RustCrypto/hashes

File: cmd/merkle-generator/src/lib.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ pub struct Sha3Algorithm(Sha3);
 
 impl Default for Sha3Algorithm {
     fn default() -> Self {
-        Self(Sha3::sha3_256())
+        Self(Sha3::v256())
```
Comment (jolestar) [inline cmd/merkle-generator/src/lib.rs:30]: Move 那边 sha3 用的库是 `sha3 = "0.9.1"`, 要不 sha3 都统一成这个库呗，这样方便维护。

Comment (nkysg) [reply cmd/merkle-generator/src/lib.rs:30]: > @nkysg Seems like you are using tiny-keccak to replace rust-crypto, but the latest commit to tini-keccak is 2 years ago. Not sure if it is maintained. When reading the codes in Move language, we noticed that they are using crate sha2 to do some hash algorithms. image
> 
> https://docs.rs/sha2/latest/sha2/ https://github.com/RustCrypto/hashes

we should using sha3 or tiny-keccak,  our HashValue using sha3 compute.
https://github.com/starcoinorg/starcoin-crypto/blob/2c95387a3e5ebfbff544e0044f039ce071125dc5/crates/diem-crypto/src/hash.rs#L199-L203
```rust
    #[cfg(not(feature = "avx512f"))]
    pub fn sha3_256_of(buffer: &[u8]) -> Self {
        let mut sha3 = Sha3::v256();
        sha3.update(buffer);
        HashValue::from_keccak(sha3)
    }
```
HashValue is port from diem-crypto

**Comment by nkysg** [comment]
> @nkysg Seems like you are using tiny-keccak to replace rust-crypto, but the latest commit to tini-keccak is 2 years ago. Not sure if it is maintained. When reading the codes in Move language, we noticed that they are using crate sha2 to do some hash algorithms. image
> 
> https://docs.rs/sha2/latest/sha2/ https://github.com/RustCrypto/hashes


we should using sha3 or tiny-keccak,  our HashValue using sha3 compute.
https://github.com/starcoinorg/starcoin-crypto/blob/2c95387a3e5ebfbff544e0044f039ce071125dc5/crates/diem-crypto/src/hash.rs#L199-L203
```rust
    #[cfg(not(feature = "avx512f"))]
    pub fn sha3_256_of(buffer: &[u8]) -> Self {
        let mut sha3 = Sha3::v256();
        sha3.update(buffer);
        HashValue::from_keccak(sha3)
    }
```
HashValue code is port from diem-crypto

File: cmd/merkle-generator/src/lib.rs (hunk 1 above)
Comment (nkysg) [reply cmd/merkle-generator/src/lib.rs:30]: It's ok, I will try to use sha3

**Review by jolestar — APPROVED** [review]

**Pull request merged**